apps/financial-aid/backend/src/app/modules/application/application.service.ts

```diff
@@ -298,14 +286,31 @@ export class ApplicationService {
     const files = await this.fileService.getAllApplicationFiles(id)
     updatedApplication?.setDataValue('files', files)
 
-    if (shouldSendEmail.includes(update.event) && update.state) {
+    if (
+      update.event === ApplicationEventType.DATANEEDED ||
+      update.event === ApplicationEventType.REJECTED ||
+      update.event === ApplicationEventType.APPROVED
+    ) {
+      const municipality = await this.municipalityService.findByMunicipalityId(
+        updatedApplication.municipalityCode,
+      )
+      const emailData = getApplicantEmailDataFromState(
```

Comment:
maybe getApplicantEmailDataFromEvent since we are working with applicationsevents? 😸

Reply:
Correct, will fix!